### PR TITLE
Chore/pd 768 delegate deletion

### PR DESCRIPTION
**Pull request by eksoverzero**
# Summary

* Remove the deletion dependency on mentions

## Updates

Remove old TravisCI config and update CircleCi to tests over multiple ruby versions using a build matrix

Screenshot from 2020-07-20 14-03-26

## Notes

Rubocop was remove (from the 2.3 build) as it was fetching the config from fuse-dev-tools. The fuse-dev-tools gem introduced gems that broke the dependency versions in the specs.

### CODE REVIEW

File: acts_as_mentionable.gemspec [hunk 1]
```diff
@@ -27,11 +27,9 @@ Gem::Specification.new do |spec|
   # Differentiate through versioned code?
   spec.add_runtime_dependency 'activerecord', '~> 4.2'
 
-  spec.add_development_dependency 'bundler', '~> 1.17'
-  spec.add_development_dependency 'database_cleaner', '~> 1.7'
   spec.add_development_dependency 'rake', '~> 10.5'
   spec.add_development_dependency 'rspec', '~> 3.8'
   spec.add_development_dependency 'rspec-its', '~> 1.2'
-  spec.add_development_dependency 'rspec_junit_formatter'
```
Comment (alastairp) [inline acts_as_mentionable.gemspec:35]: Why was the formatter removed?
Don't we use that to make CircleCI integration nicer?

Comment (eksoverzero) [reply acts_as_mentionable.gemspec:35]: For parallel tests, yes. This suite doesn't need to be run like that

File: spec/acts_as_mentionable/mentioner_spec.rb [hunk 2]
```diff
@@ -93,7 +93,9 @@
 
     before do
       allow(ActsAsMentionable::MentionsUpdater).to receive(:new) { mentions_updater }
+      # rubocop:disable RSpec/Yield
       allow(mentions_updater).to receive(:call) { |&block| block.call }
+      # rubocop:enable RSpec/Yield
```
Comment (alastairp) [inline spec/acts_as_mentionable/mentioner_spec.rb:98]: If you've removed `rubocop` from the dependencies, then you can remove these overrides.

Comment (eksoverzero) [reply spec/acts_as_mentionable/mentioner_spec.rb:98]: Rubocop is returning. I don't want to make this another dev's or tomorrows problem again

**Review by korki — APPROVED** [review]

**Review by ACronje — APPROVED** [review]

**Review by korki — APPROVED** [review]

**Pull request merged**